Update private cloud doc and some more details on sshd in prereqs (#43)
MicrosoftDocs/azure-management-docs: articles/azure-arc/servers/ssh-arc-overview.md

## Prerequisites

> - User Permissions: Owner or Contributor role assigned for the target Arc-enabled server.

Reviewer: This is not necessarily required. You need Owner or contributor to set the default endpoint and service configuration, but once those are set up you don't need that much permission. If I'm not mistaken you need "Virtual Machine Local User Login"

Reviewer: I see that is what was documented before, so I guess it means that's the requirement for setting up the feature. But it's a little confusing.

- Arc-enabled Server:
    - Hybrid Agent version: 1.31.xxxx or higher
    - SSH service ("sshd") must be enabled. 

> For Linux install `openssh-server` via a package manager. You can check if sshd is running on linux by running 

Reviewer 2: ```suggestion
For Linux, install `openssh-server` via a package manager. You can check whether sshd is running on Linux by running the following command.
```
Fixing punctuation and capitalization.

```shell
ps -aux | grep sshd
```

> On Windows see on [enable OpenSSH](/windows-server/administration/openssh/openssh_install_firstuse). You can check if ssh is installed and running by the following commands:

Reviewer 2: ```suggestion
For Windows, see [Enable OpenSSH](/windows-server/administration/openssh/openssh_install_firstuse). You can check whether ssh is installed and running by the following commands.
```
Fixing punctuation.

### Availability

> SSH access to Arc-enabled servers is currently supported in all public cloud regions supported by Arc-enabled servers and may work on private cloud environments; however functionality on private clouds is not thoroughly tested, if you encounter an error please see [troubleshooting ssh arc](https://learn.microsoft.com/azure/azure-arc/servers/ssh-arc-troubleshoot?tabs=azure-cli).

Reviewer: I don't think it's good to disclose that certain clouds are not tested. I think that creates more questions than answers and might decrease confidence in this feature. I think it's better if we just revert this to what it used to be ("SSH access to Arc-enabled servers is currently supported in all regions supported by Arc-enabled servers"), and then if someone else has issues in private cloud they can raise an issue.
